public/app/features/migrate-to-cloud/onprem/ResourceDetailsModal.tsx

```diff
@@ -9,11 +9,65 @@ interface ResourceDetailsModalProps {
   onClose: () => void;
 }
 
+function getTMessage(errorCode: string | undefined): string {
```

Comment:
Nit: I think to have type-safety on this switch, you can do something like:
```suggestion
function getTMessage(errorCode: NonNullable<MigrateDataResponseItemDto['errorCode']>): string {
```
need to import `MigrateDataResponseItemDto`, but you could remove the `default` case 🤔